generated-code-tests/src/main/kotlin/testgen/messages/message_no_fields.proto.kt

```diff
@@ -23,23 +23,15 @@ public class MessageNoFields() {
     @Serializable
     public data class SubMessageNoFieldsExtend(
       @ProtoNumber(number = 1)
-      public val type: Type = testgen.messages.MessageNoFields.Type.UNKNOWN,
+      public val type: Type? = testgen.messages.MessageNoFields.Type.UNKNOWN,
     )
   }
 
   @Serializable
   public data class SubMessageOneofFields(
     @ProtoNumber(number = 1)
     public val someValue: Int? = null,
-  ) {
-    init {
-      require(
-        listOfNotNull(
-          someValue,
-        ).size <= 1
-      ) { "Should only contain one of some_oneof." } 
-    }
-  }
```

Comment:
Hmm, I wonder whether `isSyntheticCompat` works as intended in this situation or not. I don't see a reason why this block would be removed. It is not possible to set more than 1 one ofs.